### PR TITLE
Replace SSL_PKEY_RSA_SIGN, SSL_PKEY_RSA_ENC with SSL_PKEY_RSA

### PR DESCRIPTION
**Pull request by snhenson**
##### Description of change

This replaces the SSL_PKEY_RSA_ENC and SSL_PKEY_RSA_SIGN certificate and private key types with SSL_PKEY_RSA instead. This was only ever partly implemented (SSL_PKEY_RSA_SIGN was always NULL): for signing it checked to see if RSA_PKEY_RSA_SIGN was present (which it never would be) and then tried SSL_PKEY_RSA_ENC.

### CODE REVIEW

**Comment by tmshort** [comment]
I was seriously just thinking about doing a PR for this type of change, this morning. However, I have to question the use of SSL_PKEY_RSA_SIGN name vs. SSL_PKEY_RSA_ENC name. If SIGN was always NULL, shouldn't that have been the one removed? TLSv1.2 and earlier will still use the RSA key for encrypting the PMS. Perhaps it should be referred to as SSL_PKEY_RSA?

**Comment by snhenson** [comment]
I think SSL_PKEY_RSA_ENC was used for legacy reasons: when forward secrecy ciphersuites were rarer RSA keys would most commonly be used for encrypt/decrypt. Now signing is more common.

I'd considered calling it SSL_PKEY_RSA too, we already do that with SSL_PKEY_ECC (though we only ever use that for signing now). OK I'll update the PR with that change.

**Comment by snhenson** [comment]
Updated the PR now. It's identical to the previous apart from the change to the SSL_PKEY_RSA.

File: ssl/ssl_locl.h [hunk 1]
```diff
@@ -416,7 +415,7 @@
  * TODO(TLS1.3) for now use RSA_SIGN keys for PSS
  */
 
-#define SSL_PKEY_RSA_PSS_SIGN SSL_PKEY_RSA_SIGN
```
Comment (tmshort) [inline ssl/ssl_locl.h]: Is this define (SSL_PKEY_RSA_PSS_SIGN) still needed?

Comment (tmshort) [inline ssl/ssl_locl.h]: Marked as TODO, so maybe other work then?

**Pull request closed without merging**